### PR TITLE
Simplify performMemberLookup()

### PR DESCRIPTION
**Pull request by davezarzycki**
Simplify performMemberLookup() by having the normal code path handle both simple and compound named constructors. As a part of the fallout, a redundant diagnostic can also be removed.

### CODE REVIEW

**Comment by xedin** [comment]
@swift-ci please test

**Comment by xedin** [comment]
@swift-ci please test source compatibility

**Comment by slavapestov** [comment]
It looks like the email address in your commit messages isn't associated with your GitHub account. You should add it to get credit on https://github.com/apple/swift/graphs/contributors.

File: test/Constraints/super_constructor.swift [hunk 1]
```diff
@@ -55,12 +55,12 @@ class B {
 
 // SR-2484: Bad diagnostic for incorrectly calling private init
 class SR_2484 {
-  private init() {} // expected-note {{'init()' is inaccessible due to 'private' protection level}}
-  private init(a: Int) {} // expected-note {{'init(a:)' is inaccessible due to 'private' protection level}}
+  private init() {} // expected-note {{'init' declared here}}
+  private init(a: Int) {} // expected-note {{'init' declared here}}
```
Comment (hartbit) [inline test/Constraints/super_constructor.swift]: Isn't the new diagnostic worse here?

Comment (davezarzycki) [inline test/Constraints/super_constructor.swift]: I thought that at first, but then I realized that the "new" output is just the general case diagnostics when any member lookup fails due to inaccessibility; and I didn't want to argue for changing the output of the general case. That wasn't the goal of this pull request.

**Comment by aschwaighofer** [comment]
Hi Dave,
I have reverted this commit because it broke internal bots. 
```
import Foundation
class P {
    var x : Int = 0
    var y : Int = 1
}

let dist : (P, P) -> Double = {
  (p : P, s : P)  -> Double in
    sqrt(Double((p.x-s.x)*(p.x-s.x) + (p.y-s.y)*(p.y-s.y)))
}
```
fails now with expression to complex.

**Comment by davezarzycki** [comment]
I think this regression is actually the exposure of a latent performance bug:

https://bugs.swift.org/browse/SR-5669

**Pull request merged**